protocols/kad/src/query.rs

```diff
@@ -312,15 +312,16 @@ where
                         return Async::Ready(QueryStatePollOut::CancelRpc { peer_id: peer_id.preimage() });
                     }
                     Ok(Async::NotReady) => {
+                        succeeded_counter = None;
                         active_counter += 1
                     }
                 }
             }
 
             if let QueryPeerState::Succeeded = state {
-                succeeded_counter += 1;
+                succeeded_counter.as_mut().map(|c| *c += 1);
```

Comment:
I really prefer this form:
```suggestion
                if let Some(c) = succeeded_counter.as_mut() { *c += 1; }
```

To me, the functional style is for immutability.